### ISSUE TITLE
Windows Defender detects v1.15.1 as a trojan

### ISSUE THREAD

**Issue by metaa** [Genymobile/scrcpy#1648]
- [X] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [X] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: Windows 10, 2004
 - scrcpy version: [v1.15.1](https://github.com/Genymobile/scrcpy/releases/tag/v1.15.1)
 - installation method: Windows release

**Describe the bug**
Windows Defender claims this version of `scrcpy` contains [`Trojan:Win32/Emali.A!cl`](https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?Name=trojan:win32/emali.a!cl&ThreatID=2147728322).

![image](https://user-images.githubusercontent.com/5056880/89694082-d4232a80-d910-11ea-9c9b-2385c51948e9.png)

**Comment by ndinev** [comment]
Same problem

**Comment by rom1v** [comment]
Arf, this is the same problem as #1102 :disappointed: 

Does it only block `scrcpy-noconsole.exe`, or also `scrcpy.exe`?
Is it only v1.15.1? Does it affects v1.15 or v1.14?

**Comment by rom1v** [comment]
Wow, Windows defender is triggered by v1.15.1 but not by 1.15:

- `scrcpy-win64-v1.15.1.zip`: https://www.virustotal.com/gui/file/78fba4caad6328016ea93219254b5df391f24224c519a2c8e3f070695b8b38ff/detection
- `scrcpy-win64-v1.15.zip`: https://www.virustotal.com/gui/file/dd514bb591e63ef4cd52a53c30f1153a28f59722d64690eb07bd017849edcba2/detection

The only source code difference between the two versions is: 976761956f5ff8a806a8547e7a1af9e4e7e3d77c and 633a51e9c4c64ed549a9a44f8348ded528268451 :thinking:

**Comment by Biswa96** [comment]
Windows Defender is mad. There are same issues in many repositories. Just allow it or disable it.

**Comment by metaa** [comment]
> Does it only block `scrcpy-noconsole.exe`, or also `scrcpy.exe`?

So far it only seems to complain about the `-noconsole` variant.

**Comment by SimonMarquis** [comment]
`scrcpy.exe` has been removed as well on my machine a few hours after `scrcpy-noconsole.exe`.

**Comment by quammy** [comment]
Hi, should be related - 

Just got blocked by Chrome on trying to download, getting this message:
"scrcpy-win64-v1.15.1.zipFailed - Virus detected"

V1.15 did not triggered such a message.

Great tool and thanks for your effort!!

**Comment by rom1v** [comment]
> Just got blocked by Chrome on trying to download, getting this message:
> "scrcpy-win64-v1.15.1.zipFailed - Virus detected"

Yes, I guess it's Windows defender too:

https://windowsreport.com/failed-virus-detected-chrome/

> So, you don’t have to blame Google Chrome but your own antivirus software as it is the one showing the Failed, virus detected alert message.

I'm not sure there is a simple way to report false positives (and make them confirm that the file does not contain a trojan).

**Comment by SimonMarquis** [comment]
@rom1v did you try to submit the binaries to Microsoft?

https://docs.microsoft.com/en-us/windows/security/threat-protection/intelligence/submission-guide

**Comment by quammy** [comment]
oh, ok. Just thought it can help in someway.
Anyway, with allowing it on Windows Defender I could make the download and work.
Thanks.

**Comment by rom1v** [comment]
Thank you for the link.

> **Can I submit a sample without signing in?**
>
> Yes, you many submit a file as an anonymous home customer. You will get a link to a webpage where you can view the status of the submission.

Contrary to what this page says, as soon as I click on submit, it requires to be logged with some account.

I just created an account and submitted "as a software developer" the file `scrcpy-noconsole.exe`.

**Comment by SimonMarquis** [comment]
> Contrary to what this page says, as soon as I click on submit, it requires to be logged with some account.

Yeah, I experienced the same frustration... It might have been available when they first launched the portal.
Anyway, let's hope they can give us more info on why they flagged it, and maybe unflag it.

**Comment by AndroidDeveloperLB** [comment]
Is it ok to use v1.15.0 instead? Is it much of a difference ? I've noticed both were published very near each other...

**Comment by KeronCyst** [comment]
Can confirm:

![image](https://user-images.githubusercontent.com/13827055/89741494-a6400200-da46-11ea-8675-27eb4c819bb2.png)

**Comment by SimonMarquis** [comment]
> Is it ok to use v1.15.0 instead? Is it much of a difference ? I've noticed both were published very near each other...

Here is the diff: [v1.15...v1.15.1](https://github.com/Genymobile/scrcpy/compare/v1.15...v1.15.1)

v1.15.1 was a hotfix for #1643

**Comment by AndroidDeveloperLB** [comment]
@SimonMarquis Doesn't seem so important, no?

**Comment by metaa** [comment]
It's probably Defender detecting some kind of evil in one of the SHA sums.
I don't know too much about antivirus though 🤔

**Comment by SimonMarquis** [comment]
@AndroidDeveloperLB no it doesn't seem that important and if v1.15 works for you, I guess you can keep it for now.
@metaa we'll see what the submission of the binary to Microsoft will tell us. 🤷

**Comment by rom1v** [comment]
For some reason, v1.16 is not detected as a trojan: https://www.virustotal.com/gui/file/3f30dc5db1a2f95c2b40a0f5de91ec1642d9f53799250a8c529bc882bc0918f0/detection

:tada:

The problem is not "solved", but now that it seems ok with the latest release, I close this issue since it's a duplicate of #1102.

**Comment by rom1v** [comment]
> I just created an account and submitted "as a software developer" the file `scrcpy-noconsole.exe`.

![analysis](https://user-images.githubusercontent.com/543275/90079121-462bb300-dd07-11ea-8c98-272f2284f1ee.png)

> **Analyst comments:**
>
> The submitted files do not meet our criteria for detection. No detection will be added for these files.
>
> More detailed information about the approach and criteria categories currently used by the Microsoft researchers are available here:
> https://www.microsoft.com/en-us/wdsi/antimalware-support/malware-and-unwanted-software-evaluation-criteria
> 
> Thank you for contacting Microsoft.

I'm not sure to understand why it does not meet their criteria. And in that report, they say "no malware detected" :thinking:

**Comment by metaa** [comment]
> I'm not sure to understand why it does not meet their criteria. And in that report, they say "no malware detected" 🤔

Microsoft being Microsoft.
I think as long as it doesn't occur with the newer versions, we can safely drop this case.

**Comment by will9183** [comment]
@rom1v I rescanned the file through VirusTotal and it says no malware detected from Microsoft anymore, but it is still detected as malware with SecureAge APEX and Rising, but that is not a huge issue. https://www.virustotal.com/gui/file/d07c28b00f59281fc0db77c8ad10fc8f7d6effbd4371331068f62ce2188e6f07/detection

**Comment by AndroidDeveloperLB** [comment]
What does it mean "meet our criteria for detection" ? 
There is a criteria to detect viruses?

**Comment by AzureRaptor** [comment]
> > The submitted files do not meet our criteria for detection. No detection will be added for these files.
> > More detailed information about the approach and criteria categories currently used by the Microsoft researchers are available here:
> > https://www.microsoft.com/en-us/wdsi/antimalware-support/malware-and-unwanted-software-evaluation-criteria
> > Thank you for contacting Microsoft.
> 
> I'm not sure to understand why it does not meet their criteria. And in that report, they say "no malware detected" 🤔

I believe what they mean is those files don't meet their criteria for detection _as malware._

**Comment by AndroidDeveloperLB** [comment]
But it's their AV app that reported it as such...
I don't understand.
